library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -275,12 +350,7 @@ impl Command {
 
     // Only support platforms for which posix_spawn() can return ENOENT
     // directly.
-    #[cfg(any(
-        target_os = "macos",
-        target_os = "freebsd",
-        all(target_os = "linux", target_env = "gnu"),
-        all(target_os = "linux", target_env = "musl"),
-    ))]
+    #[cfg(any(target_os = "macos", target_os = "freebsd",))]
```

Comment:
I'm not sure why this was changed. I added `self.make_pidfd` to the conditions for the early return below.